src/pm/hydra/include/hydra.h

```diff
@@ -620,7 +620,7 @@ HYD_status HYDU_sock_cloexec(int fd);
 #define HYDU_MALLOC_OR_JUMP(p, type, size, status)                      \
     {                                                                   \
         (p) = (type) MPL_malloc((size), MPL_MEM_PM);                 \
-        if ((size) && (p) == NULL)                                      \
+        if ((size > 0) && ((p) == NULL))                                \
```

Comment:
I used `(size) != 0` in #3208. Doesn't it preserve the original logic better?

Reply:
Yeah, that's true. Though `size <= 0` is probably invalid.

Reply:
Yeah, I wondered the same thing but was lazy to see if signed variable is passed as `size`. At L626 `size` is cast to `int` (although this might be wrong too).